src/swap/handle_swap_sign_transaction.c

```diff
@@ -136,20 +136,22 @@ bool swap_check_validity() {
         os_sched_exit(0);
     }
 
-    if (G_context.tx_info.transaction.hints_type != TRANSACTION_COMMENT) {
-        PRINTF("Wrong operation %d\n", G_context.tx_info.transaction.hints_type);
+    if ((G_context.tx_info.transaction.hints_type == TRANSACTION_COMMENT) &&
+        (G_context.tx_info.transaction.hints_len != 0)) {
+        PRINTF("Hint length %d refused\n", G_context.tx_info.transaction.hints_len);
         io_send_sw(SW_SWAP_FAILURE);
         // unreachable
         os_sched_exit(0);
-    } else if (G_context.tx_info.transaction.hints_len != 0) {
-        PRINTF("Hint length %d refused\n", G_context.tx_info.transaction.hints_len);
+    } else if ((G_context.tx_info.transaction.hints_type != TRANSACTION_COMMENT) &&
+               (G_context.tx_info.transaction.hints_type != TRANSACTION_TRANSFER_JETTON)) {
+        PRINTF("Wrong operation %d\n", G_context.tx_info.transaction.hints_type);
         io_send_sw(SW_SWAP_FAILURE);
         // unreachable
         os_sched_exit(0);
-    } else {
-        PRINTF("Valid operation %d\n", G_context.tx_info.transaction.hints_type);
     }
```

Comment:
I'm not certain but I think we should be more strict in this check
If we are swapping jettons  (currency is not TON) == we require a TRANSACTION_TRANSFER_JETTON
Otherwise we forbid it